packages/babel-types/src/builders/generated/index.ts

```diff
@@ -2,28 +2,45 @@
  * This file is auto-generated! Do not modify it directly.
  * To re-generate run 'make build'
  */
-import validateNode from "../validateNode.ts";
+import * as _validate from "../../validators/validate.ts";
 import type * as t from "../../index.ts";
 import deprecationWarning from "../../utils/deprecationWarning.ts";
+import { NODE_FIELDS, type FieldOptions } from "../../definitions/utils.ts";
+import astOrderData from "../../../ast-order-data.json" with { type: "json" };
+
+const { validateInternal: validate } = _validate;
+
+const _data: FieldOptions[][] = [];
+for (const [type, fields] of astOrderData as [string, []][]) {
+  const fieldOptions = NODE_FIELDS[type];
+  _data.push(fieldOptions ? fields.map(field => fieldOptions[field]) : null);
+}
+
 export function arrayExpression(
   elements: Array<null | t.Expression | t.SpreadElement> = [],
 ): t.ArrayExpression {
-  return validateNode<t.ArrayExpression>({
+  const node: t.ArrayExpression = {
     type: "ArrayExpression",
     elements,
-  });
+  };
+  validate(_data[0][0], node, "elements", elements, true);
+  return node;
 }
 export function assignmentExpression(
   operator: string,
   left: t.LVal | t.OptionalMemberExpression,
   right: t.Expression,
 ): t.AssignmentExpression {
-  return validateNode<t.AssignmentExpression>({
+  const node: t.AssignmentExpression = {
     type: "AssignmentExpression",
     operator,
     left,
     right,
-  });
+  };
+  validate(_data[1][0], node, "operator", operator);
+  validate(_data[1][1], node, "left", left, true);
+  validate(_data[1][2], node, "right", right, true);
```

Comment:
Can we hard-code numbers here, instead of using `_data[1][2]`?

Reply:
The data in the array is like this.
```
export type FieldOptions = {
  default?: string | number | boolean | [];
  optional?: boolean;
  deprecated?: boolean;
  validate?: Validator;
};
```

Reply:
Or even just `validate(node.type, node, "operator", operator)`, and then `validate` internally can get `NODE_FIELDS[node.type]["operator"]`.

Reply:
This avoids a lookup in `NODE_FIELDS`.
Also array access may be faster than `.["operator"]`.

Reply:
It seems that the performance data has little impact, I will open a new PR to try.